firmware/controllers/algo/dynoview.cpp

```diff
@@ -0,0 +1,181 @@
+/*
+ * @file dynoview.cpp
+ *
+ * @date Nov 29, 2020
+ * @author Alexandru Miculescu, (c) 2012-2020
+ */
+
+#include "engine.h"
+
+#if EFI_DYNO_VIEW
+#include "dynoview.h"
+#include "vehicle_speed.h"
+
+static Logging *logger;
+
+#if EFI_TUNER_STUDIO
+#include "tunerstudio_outputs.h"
+extern TunerStudioOutputChannels tsOutputChannels;
+#endif /* EFI_TUNER_STUDIO */
+
+EXTERN_ENGINE;
+
+DynoViewBase dynoInstance;
```

Comment:
probably doesn't need to be `XxxBase` since nobody inherits it, it can just be `DynoView` (the other option is to make virtual functions for the hardware-sensitive parts so they can be mocked in tests, then it should be `XxxBase`)